drivers/sensor/ti/tmp114/tmp114.c

```diff
@@ -155,8 +165,41 @@ static int tmp114_attr_get(const struct device *dev, enum sensor_channel chan,
 	return 0;
 }
 
+static int tmp114_attr_set(const struct device *dev,
+			   enum sensor_channel chan,
+			   enum sensor_attribute attr,
+			   const struct sensor_value *val)
+{
+	int16_t value;
+	int rc;
+
+	if (chan != SENSOR_CHAN_AMBIENT_TEMP) {
+		return -ENOTSUP;
+	}
+
+	switch (attr) {
+	case SENSOR_ATTR_OVERSAMPLING:
+		/* Enable the AVG in tmp114. The chip will do 8 avg of 8 samples
+		 * to get a more accurate value.
+		 */
+		rc = tmp114_reg_read(dev, TMP114_REG_CFGR, &value);
+		if (rc < 0) {
+			return rc;
+		}
+		value = value & ~TMP114_AVG_MASK;
```

Comment:
Can WRITE_BIT be used here?
https://docs.zephyrproject.org/latest/doxygen/html/group__sys-util.html#ga23a900e882ecb48455e70f01fd45b246